File: its-my-training-day/leetcode-top-75/questions/arrays-hashing/hash-map/longest-consecutive/solution.py
import os
import sys
from typing import List, Dict, Optional, Tuple

ROOT_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), "../../../../"))
if ROOT_DIR not in sys.path:
    sys.path.append(ROOT_DIR)

from shared.python.ds import ArrayList, ListNode, Stack, Queue, TreeNode, MinHeap, MaxHeap

def longest_consecutive(nums: List[int]) -> int:
    seen = {}
    i = 0
    while i < len(nums):
        seen[nums[i]] = True
        i += 1
    longest = 0
    i = 0
    while i < len(nums):
        val = nums[i]
        if (val - 1) not in seen:
            length = 1
            current = val + 1
            while current in seen:
                length += 1
                current += 1
            if length > longest:
                longest = length
        i += 1
    return longest

File: its-my-training-day/leetcode-top-75/questions/arrays-hashing/hash-map/longest-consecutive/solution.ts
import {
  ArrayList,
  Stack,
  Queue,
  ListNode,
  TreeNode,
  MinHeap,
  MaxHeap,
} from "../../../../shared/typescript/ds";

export const longestConsecutive = (nums: number[]): number => {
  const seen: Record<number, boolean> = {};
  for (let i = 0; i < nums.length; i += 1) seen[nums[i]] = true;
  let longest = 0;
  for (let i = 0; i < nums.length; i += 1) {
    const val = nums[i];
    if (seen[val - 1] !== true) {
      let length = 1;
      let current = val + 1;
      while (seen[current]) {
        length += 1;
        current += 1;
      }
      if (length > longest) longest = length;
    }
  }
  return longest;
};
